### DISCUSSION TITLE
Force Refresh Button on Live Preview

### DISCUSSION THREAD

**Discussion by richhayler** [craftcms/cms#11160] [Ideas] [answered]
Please could we have a "force refresh" button added to the Live Preview Toolbar to force a refresh of the page currently being previewed.

**Use case**
If a page preview doesn't load properly, e.g. due to a template error (which is then subsequently fixed while the preview pane is open), then the page preview could easily be re-rendered on demand without having to close and then reopen the page preview pane to refresh the page.

<img width="721" alt="CleanShot 2022-05-09 at 13 17 40@2x" src="https://user-images.githubusercontent.com/2505643/167335114-d2530271-b74a-4a7b-8253-e5f2baf50f4c.png">

Also if the option to auto refresh the preview pane was unchecked (as part of configuring the preview target), this would provide the option to refresh the preview manually. 

![7883F358-363A-4402-B68B-3FFA219A91C6](https://user-images.githubusercontent.com/2505643/167511426-bbaeaf01-e583-4274-b5a2-2ba89270306d.jpeg)

**Comment by brandonkelly** [comment, marked as answer]
There already was a “Refresh” button, for targets where “Auto-refresh” was unchecked.

Just tweaked it so it’ll show regardless of the “Auto-refresh” setting, though, for Craft 4.1 (1716c1ed8c55e84ed50179389ba7bb212459613c).

**Reply by richhayler** [reply]
Thanks Brandon :)

**Reply by brandonkelly** [reply]
Craft 4.1 is out with this ✨
